mentionjobiter.go

```diff
@@ -61,20 +63,33 @@ func (i *mentionJobIter) initIter() error {
 	return nil
 }
 
-// getEndpoint obtains the next Job from the queue and decodes the mention on it.
-// If success, the endpoint into the mention is returned. Also the job itself is
+// getEndpoints obtains the next Job from the queue and decodes the mention on it.
+// If success, ALL the endpoints into the mention are returned. Also the job itself is
 // returned, to be able to send back the ACK.
-func (i *mentionJobIter) getEndpoint() (string, *queue.Job, error) {
-	j, err := i.iter.Next()
+func (i *mentionJobIter) getEndpoints() (a []string, j *queue.Job, err error) {
+	j, err = i.iter.Next()
 	if err != nil {
-		return "", nil, err
+		return
```

Comment:
you can set directly output variables, in this case `(a []string, j *queue.Job, err error)`